apps/gallery/test/marionette/activity_test.js

```diff
@@ -0,0 +1,156 @@
+/* global require, marionette, setup, __dirname */
+'use strict';
+
+var Gallery = require('./lib/gallery.js'),
+    GalleryActivityTester = require('./lib/galleryactivitytester.js'),
+    System = require('./lib/system.js'),
+    assert = require('assert');
+
+marionette('Gallery Activity Tests', function() {
+
+  var client, apps = {};
+
+  apps[GalleryActivityTester.ORIGIN] =
+    __dirname + '/apps/galleryactivitytester';
+
+  client = marionette.client({
+    profile: {
+      prefs: {
+        'device.storage.enabled': true,
+        'device.storage.testing': true,
+        'device.storage.prompt.testing': true
+      },
+      apps: apps
+    }
+  });
+
+  var galleryApp, activityTesterApp, system;
+  var imageInfo = {
+      name: 'pictures/firefoxOS.png',
+      type: 'image/png',
+      size: '52458'
+    };
+
+  setup(function() {
+    // Remove all files in temp device storage.
+    client.fileManager.removeAllFiles();
+    // Add file into the pictures directory
+    client.fileManager.add({
+      type: 'pictures',
+      filePath: 'test_media/Pictures/firefoxOS.png'
+    });
+    galleryApp = new Gallery(client);
+    activityTesterApp = GalleryActivityTester.create(client);
+    system = new System(client);
+  });
+
+  test('pick and open image using gallery pick and open activity', function() {
+    activityTesterApp.launch();
+    // Initiate pick activity by clicking pick image button
+    activityTesterApp.tapPickImageButton();
+
+    // Tap Gallery button from Systm app Action Menu
+    system.menuOptionButton('Gallery').tap();
+
+    // We've switched to the system app, and we're calling
+    // switchToApp to switch to gallery
+    system.switchToApp(Gallery.ORIGIN);
```

Comment:
The tap() call above should cause Gallery to be launched for the pick activity. Does marionette's switchToApp() implicitly wait for that app to start running before switching to it?  I just want to make sure that switchToApp() isn't going to fail because of timing issues or launch a Gallery instance on its own.  I think I asked you that before though, so I assume this works.  Maybe add a comment saying that it waits.

(And also: do you need system.switchToApp(), or could you just do client.apps.switchToApp(Gallery.ORIGIN) from here?)